server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3033,22 +2991,33 @@ private static boolean isQueued(@Nullable ShardSnapshotStatus status) {
         private final ClusterState currentState;
 
         // updates outstanding to be applied to existing snapshot entries
-        private final List<ShardSnapshotUpdate> unconsumedUpdates;
+        private final Map<String, List<ShardSnapshotUpdate>> unconsumedUpdates;
```

Comment:
We don't actually consume these updates (we used to before introducing `SnapshotShardsUpdateContext` it seems) we just iterate over them. So maybe `updatesByRepo` would be a better name?

Reply:
++